lib/tasks/util.rake

```diff
@@ -33,6 +28,27 @@ namespace :redmine_ip_filter do
       puts addresses.map {|address| "ADD\t#{address}"}
     end
 
+    desc 'Export the allowed IP addresses with comments'
+    task :export => :environment do
+      puts FilterRule.find_or_default.allowed_ips_with_comments
+    end
+
+    desc 'Import the allowed IP addresses with comments'
+    task :import => :environment do
+      filter_rule = FilterRule.find_or_default
+      import_address_with_comments = ENV['IMPORT']
+      if import_address_with_comments.blank?
+        abort 'IP addresses to add must be specified with IMPORT environment variable'
+      else
+        filter_rule.allowed_ips = ([filter_rule.allowed_ips_with_comments, import_address_with_comments]).join("\n")
+      end
```

Comment:
Hello @takenory . Thank you for the pull request.

In my review at https://github.com/redmica/redmine_ip_filter/pull/36#pullrequestreview-1853119678, I made a mistake in my suggestion.

Right now, the Import feature adds to the current allowed IP addresses. But, I think it should replace the current allowed IP addresses with the imported ones.

```suggestion
      if import_address_with_comments.blank?
        abort 'IP addresses to import must be specified with IMPORT environment variable'
      else
        filter_rule.allowed_ips = import_address_with_comments
      end
```

My suggestion has a downside. Since it's not adding, if setting the allowed IP addresses fails, we would have to run the import again or manually add remote addresses. But, since we can run rake tasks, I think we can handle it.

I would appreciate your feedback on this matter.

Reply:
@maeda-m  Thx for suggetion.

I thought it was debatable whether the behavior of import should be "replace" or not.
When importing issues from CSV in Redmine, the imported data behaves as "appended", not "replaced".

If the risk is that a successful import with incorrect data will "replace" the original IP address settings (and in some cases make them unrecoverable), I think it is safer to behave as an "addition".